### PR TITLE
Add to onboarding reproduction logs

### PR DESCRIPTION
**Pull request by Samantha-Zhan**
Details on my setup (e.g., operating system, environment and configuration, etc.):
- Windows Ubuntu 24.04.1 LTS (WSL2)
- Java 21
- conda 24.1.2

The setup was pretty intuitive and tasks were easy to follow; since I've completed the tasks spanning multiple weeks, there have been a few weird errors that ended up was caused by not pulling the newest version or forgetting to call `conda activate pyserini`.

*nit*: had to import  `LuceneIndexReader` instead of `IndexReader` [(see doc)](https://github.com/castorini/pyserini/blob/master/docs/conceptual-framework2.md)

### CODE REVIEW

**Review by lintool — APPROVED** [review]

**Comment by lintool** [comment]
> _nit_: had to import `LuceneIndexReader` instead of `IndexReader` [(see doc)](https://github.com/castorini/pyserini/blob/master/docs/conceptual-framework2.md)

Thanks for catching. Can you add this fix to your PR? I'll then merge.

**Comment by Samantha-Zhan** [comment]
> > _nit_: had to import `LuceneIndexReader` instead of `IndexReader` [(see doc)](https://github.com/castorini/pyserini/blob/master/docs/conceptual-framework2.md)
> 
> Thanks for catching. Can you add this fix to your PR? I'll then merge.

Fix added!

File: docs/conceptual-framework.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ Conceptually, you've computed the BM25 document vector of every document in the
 With the `IndexReader` class in Pyserini, you can materialize (i.e., reconstruct) the BM25 document vector for a particular document:
 
 ```python
-from pyserini.index.lucene import IndexReader
+from pyserini.index.lucene import LuceneIndexReader as IndexReader
 import json
```
Comment (lintool) [inline docs/conceptual-framework.md:138]: nah... kill the `as` and just change `LuceneIndexReader` everywhere please?

**Comment by lintool** [comment]
See my comments?

**Comment by Samantha-Zhan** [comment]
Fix added!

**Pull request merged**